### PR TITLE
MAYA-124365 test prim undo delete vs display layer

### PR DESCRIPTION
**Pull request by pierrebai-adsk**

### CODE REVIEW

File: test/lib/ufe/testDisplayLayer.py [hunk 1]
```diff
@@ -304,8 +304,55 @@ def testDisplayLayerItemDelete(self):
...
+
+        self._testLayerFromPath(self.CUBE1, self.LAYER1)
+        self._testLayerFromPath(self.SPHERE1, self.LAYER1)
+
+        # Delete the Sphere and make sure it is removed from the layer.
+        # To allow undo to work, we need to have an undo block open when
+        # executing the command. In Maya, using the UI, commands always
+        # have an undo chunk. In scripting, we must create an explicit one.
+        cmds.undoInfo(state=1)
+        cmds.undoInfo(openChunk=1)
```
Comment (pierrebai-adsk) [inline test/lib/ufe/testDisplayLayer.py:342]: When issuing commands in scripts, we need to create undo check by hand if we want them... display layer restoration only work when we have a undo chunk.

File: test/lib/ufe/testDisplayLayer.py [hunk 2]
```diff
@@ -304,8 +304,51 @@ def testDisplayLayerItemDelete(self):
...
+        # Delete the Sphere and make sure it is removed from the layer.
+        cmds.delete(self.SPHERE1)
+
+        verifyInLayer({ self.CUBE1 }, { self.NEW_SPHERE1, self.SPHERE1 })
+
+        # Must issue two undo because delete spawed another command
+        # That updated the display layer. The alternative would be to
+        # open an explicit undo chunk to combine both commands into one undo.
+        cmds.undo()
+        cmds.undo()
```
Comment (ppt-adsk) [inline test/lib/ufe/testDisplayLayer.py:346]: This feels very awkward.  I know you've mentioned that in the interactive application all commands are chunked, so to test what the interactive application is doing I would follow your suggestion and open an explicit undo chunk before line 338.

**Review by ppt-adsk — CHANGES_REQUESTED** [review]

File: test/lib/ufe/testDisplayLayer.py [hunk 3]
```diff
@@ -304,8 +304,51 @@ def testDisplayLayerItemDelete(self):
             self.assertFalse(defaultLayer.contains(self.SPHERE1))
         self.assertFalse(layer1.contains(self.SPHERE1))
 
-        # TODO - test the undo/redo for delete of Ufe item in Layer.
-        # This currently doesn't work.
+    @unittest.skipUnless(mayaUtils.ufeSupportFixLevel() >= 6, "Requires Display Layer Ufe item delete undo fix.")
```
Comment (ppt-adsk) [inline test/lib/ufe/testDisplayLayer.py:307]: Is the fix level still O.K.?

Comment (pierrebai-adsk) [reply test/lib/ufe/testDisplayLayer.py:307]: Yes, it is still 6 for that fix.

**Comment by pierrebai-adsk** [comment]
Another PF where the interactive Linux builds failed to start. The interactive Linux instances must be down. Unrelated to my changes, all other builds passed.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**